### PR TITLE
Feature/deep linking

### LINKED ISSUE #22: Deep linking
opening a url in the app

### PR DESCRIPTION
**Pull request by DeD1rk**
Migrate to Router/Navigator 2.0 to support Deep Linking.
Add deep linking configuration for 'thaliapp://thalia.nu/' and 'https://thalia.nu/' links.
This should work, but somehow it does not. Deep Links open the app, but the url is 
not actually being parsed by the RouteInformationParser. The parser is only called 
with "/" upon starting the app.

The deep linking has to be fixed. 
After that the routerDelegate should be improved a bit, as described by TODO comments.

Closes #22

### CODE REVIEW

File: lib/navigation.dart [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  static MyRouterDelegate of(BuildContext context) {
+    RouterDelegate delegate = Router.of(context).routerDelegate;
+    assert(delegate is MyRouterDelegate, 'Delegate type must match.');
+    return delegate as MyRouterDelegate;
+  }
+
+  MyRouterDelegate() : navigatorKey = GlobalKey<NavigatorState>();
+
+  // TODO: keep app state
+  List<Page> _stack = [MaterialPage(child: SplashScreen())];
```
Comment (KiOui) [inline lib/navigation.dart:23]: We should first clear all TODO's before merging with master.

File: lib/navigation.dart [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  /// Adds a page to the top of the stack.
+  void push(Page page) {
+    _stack.add(page);
+    notifyListeners();
+  }
+
+  /// Removes the top of the stack.
+  void pop() {
+    // TODO: allow return values? This might still be handled by Navigator.
```
Comment (KiOui) [inline lib/navigation.dart:57]: Here as well

File: lib/navigation.dart [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    String path = uri.path;
+    List<String> segments = uri.pathSegments;
+
+    // Handle "/".
+    if (uri.pathSegments.length == 0)
+      return [MaterialPage(child: SplashScreen())];
+
+    // TODO: How is being logged out handled? Probably shouldn't just open the
+    // right place, unless we handle logged out redirection inside each screen,
+    // or in a Builder wrapper inside pages.
```
Comment (KiOui) [inline lib/navigation.dart:92]: And here as well

File: lib/providers/auth_provider.dart [hunk 4]
```diff
@@ -64,9 +64,11 @@ class AuthProvider extends ChangeNotifier {
   logOut() {
     _helper.disconnect();
     _status = AuthStatus.SIGNED_OUT;
+    notifyListeners();
   }
 
   Future<String> logIn() async {
+    // TODO: handle unexpected behaviour (e.g. clicking cancel)
     try {
```
Comment (KiOui) [inline lib/providers/auth_provider.dart:72]: There is a TODO here as well

**Review by KiOui — APPROVED** [review]

File: lib/providers/auth_provider.dart (hunk 4 above)
Comment (DeD1rk) [reply lib/providers/auth_provider.dart:72]: This is not related to this issue (and I think more of an issue for the oath_client plugin), so I'll leave it in

**Pull request merged**